crates/oxc_ast/src/ast/js.rs

```diff
@@ -1324,6 +1324,8 @@ pub struct Class<'a> {
     pub r#abstract: bool,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub decorators: Option<Vec<'a, Decorator<'a>>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub modifiers: Option<Vec<'a, Modifier>>,
     pub declare: bool,
```

Comment:
I just realized these `declare` and `r#abstract` can be removed? Or does the modifiers only keep the bad modifiers?

Reply:
I see, tsc has this comment: The following properties are used only to report grammar errors.

Should we add this comment as well?

Reply:
What's the motivation for using `Option<Vec>>` over `Vec` (and skip serializing if `Vec::is_empty`)?

Reply:
Yes we could do that. 

Also I was investigating a bit and found the `parse_class` function is also called in other places where the modifers are not properly parsed beforehand. So these two parameters might not always be consistent with `modifiers` now.

Reply:
> What's the motivation for using `Option<Vec>>` over `Vec` (and skip serializing if `Vec::is_empty`)? What's the semantic of `Some(vec![])`, or is that "invalid"?

@MichaReiser The `Option` is used for serialization purposes, because they don't exist in JavaScript.

Yes `Some(vec![])` is an invalid case.

Reply:
> @MichaReiser The Option is used for serialization purposes, because they don't exist in JavaScript.

I'm not sure if I understand. Or I don't understand how it's different from using a `Vec` with 

```rust
#[serde(skip_serializing_if = "Vec::is_empty")]
modifiers: Vec<...>
```

Reply:
> > @MichaReiser The Option is used for serialization purposes, because they don't exist in JavaScript.
> 
> I'm not sure if I understand. How is it different from using a `Vec` with
> 
> ```rust
> #[serde(skip_serializing_if = "Vec::is_empty")]
> modifiers: Vec<...>
> ```
> 
> Using an empty `vec` doesn't get serialized the same way as `None`. A vector with at least one element gets serialized, the same as `Some`.

Ah! Super good catch! I was in the mentality of `None` = JavaScript and didn't think more deeply. I'll create an issue for this.